subsys/bluetooth/audio/bap_stream.c

```diff
@@ -256,6 +305,37 @@ int bt_bap_stream_send(struct bt_bap_stream *stream, struct net_buf *buf,
 	return bt_iso_chan_send(bt_bap_stream_iso_chan_get(stream),
 				buf, seq_num, ts);
 }
+
+int bt_bap_stream_get_tx_sync(struct bt_bap_stream *stream, struct bt_iso_tx_info *info)
+{
+	struct bt_iso_chan *iso_chan;
+
+	CHECKIF(stream == NULL) {
+		LOG_DBG("stream is null");
+
+		return -EINVAL;
+	}
+
+	CHECKIF(info == NULL) {
+		LOG_DBG("info is null");
+
+		return -EINVAL;
+	}
+
+	if (!bt_bap_stream_can_send(stream)) {
+		LOG_DBG("Stream is not configured for TX");
+
+		return -EINVAL;
+	}
+
+	iso_chan = bt_bap_stream_iso_chan_get(stream);
+	if (iso_chan == NULL) {
+		LOG_DBG("Could not get iso channel from stream %p", stream);
+		return -EINVAL;
+	}
+
+	return bt_iso_chan_get_tx_sync(iso_chan, info);
```

Comment:
Do I understand it correctly?
if `bt_bap_stream_can_send` is false , this function early returns `-EINVAL` so why `bool can_send;` is needed? Won't it be always true when the `bt_bap_stream_get_tx_sync` calls `bt_iso_chan_get_tx_sync`?

Reply:
> if bt_bap_stream_can_send is false , this function early returns -EINVAL
Correct

> Won't it be always true when the bt_bap_stream_get_tx_sync calls bt_iso_chan_get_tx_sync?

Yes

> so why bool can_send; is needed?

The addition of `can_send` in the EP info is just a shorthand way of an application getting information about whether the stream/endpoint can send. It is not related to this function, so I am not sure what your question is. I could answer it as it is not needed as we could just have that check internally, but it may be a useful thing for applications to have (similar to the `bt_iso_info.can_send` and `bt_iso_info.can_recv` fields).